src/components/data/RenameTableModal.js

```diff
@@ -16,12 +19,46 @@ const RenameTableModal = ({ onDismiss, onUpdateSuccess, namespace, name, selecte
   // State
   const [newName, setNewName] = useState('')
   const [renaming, setRenaming] = useState(false)
+  const [renameSetTable, setRenameSetTable] = useState(false)
+
+  const setTableExists = _.includes(`${selectedDataType}_set`, _.keys(entityMetadata))
 
   const {
     getAllSavedColumnSettings,
     updateAllSavedColumnSettings
   } = useSavedColumnSettings({ workspaceId: { namespace, name }, entityType: selectedDataType, entityMetadata })
 
+  const handleTableRename = async ({ oldName, newName }) => {
+    await Ajax().Metrics.captureEvent(Events.workspaceDataRenameTable, { oldName, newName })
+    await Ajax().Workspaces.workspace(namespace, name).renameEntityType(oldName, newName)
+  }
+
+  const handleColumnSettingsRename = async ({ oldName, newName, renameSetTable }) => {
+    // Move column settings to new table(s)
```

Comment:
I realize the below is not beautiful, but the update to these column settings needed to happen in the same call to avoid calling the `useSavedColumnSettings` react hook in a function